packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -587,3 +593,45 @@ export class TelemetryNullLogger implements ITelemetryLogger {
     public sendErrorEvent(event: ITelemetryErrorEvent, error?: any): void {}
     public sendPerformanceEvent(event: ITelemetryPerformanceEvent, error?: any): void {}
 }
+
+/**
+ * Take in a event object, stringify any fields that are non-primitives, and return the new event object.
+ * @param event - Event with fields you want to stringify.
+ */
+ function convertToBaseEvent(event: ITelemetryGenericEventExt): ITelemetryBaseEvent {
+    const newEvent = {
+        ...event
+    };
+    for (const key of Object.keys(event)) {
+        const filteredEventVal = convertToBasePropertyType(event[key], key);
+        if (filteredEventVal !== null) {
+            newEvent[key] = filteredEventVal;
+        }
+    }
+    return newEvent as ITelemetryBaseEvent;
+}
+
+/**
+ * Takes in parameter, if parameter is of primitive type, return the original value.
+ * If parameter is an array, stringify then return the result.
+ * @param x - parameter passed to validate/filter
+ */
+function convertToBasePropertyType(x: any, key: string): TelemetryEventPropertyType | null {
+    switch (typeof x) {
+        case "string":
+        case "number":
+        case "boolean":
+        case "undefined":
+            return x;
+        default:
```

Comment:
Add case for "object" and do all this in there.  Then in the default case return `INVALID PROPERTY (typed as ${typeof x})` or something